src/score.py

```diff
@@ -77,4 +78,89 @@ def compute_weighted_acc(da_fc, da_true):
     )
     return acc
 
+def compute_weighted_meanspread(prediction,mean_dims=xr.ALL_DIMS):
+    """
+    prediction: xarray. Coordinates: time, forecast_number, lat, lon. Variables: z500, t850
+    time: Let there be I initial conditions
+    forecast_number: For each initial condition, let there be N forecasts
+
+    #mean variance
+    #1. for each input i, for each gridpoint, find variance among all N forecasts for that single input i
+    #2. for each input i, find latitude-weighted average of all the lat*lon points
+    #3. find average of all I inputs. take square root
+    """
+    #ToDO: add assert condition to check for input size. Alternatively, if input does not have 'time' then add it as dimension
+    var1=prediction.var('forecast_number') #is it okay to use a specific name like this (?)
+    weights_lat = np.cos(np.deg2rad(var1.lat))
+    weights_lat /= weights_lat.mean()
+    mean_spread= np.sqrt((var1*weights_lat).mean(mean_dims))
+    #var2 =  (var1*weights_lat).mean(dim={'lat','lon'})
+    #var2=var1.mean(dim={'lat','lon'}) #without weighted latitude.
+    #mean_spread=np.sqrt(var2.mean('time'))
+    
+    if type(mean_spread) is xr.Dataset:
+        mean_spread = mean_spread.rename({v: v + '_mean_spread' for v in mean_spread})
+    else: # DataArray
+        mean_spread.name = error.name + '_mean_spread' if not error.name is None else 'mean_spread'
+    return mean_spread
+
+def crps_score(observation,prediction,forecast_axis,mean_dims=xr.ALL_DIMS): 
```

Comment:
I think it would be nicer to check for the axis dimension by name, see http://xarray.pydata.org/en/stable/generated/xarray.DataArray.get_axis_num.html

This means, of course, that we have to be consistent with the naming. Check my comment above.